activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -30,7 +30,7 @@ def self.references(attributes)
         if value.is_a?(Hash)
           result << Arel.sql(key)
         elsif key.include?(".")
-          result << Arel.sql(key.split(".").first)
+          result << Arel.sql(key.split(".")[0..-2].join("."))
```

Comment:
I think `[0..-2]` is a bit hard to grasp. Given we do this in two places it might make sense to extract a small private method here, e.g.

```ruby
def split_column_name(key)
  *table, column = key.split(".")
  [table.join("."), column]
end
```

Reply:
At a glance I feel `key.split(".")[0..-2].join(".")` is a bit expensive. How about `String#[]` like `idx = key.rindex("."); key[0, idx]`?

Reply:
I was a little meh on the split as well. I like the rindex idea, can rewrite that way.